pkg/services/ngalert/state/persister_sync.go

```diff
@@ -0,0 +1,111 @@
+package state
+
+import (
+	"context"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	"github.com/grafana/dskit/concurrency"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	ngModels "github.com/grafana/grafana/pkg/services/ngalert/models"
+)
+
+type SyncStatePersister struct {
+	log   log.Logger
+	store InstanceStore
+	// doNotSaveNormalState controls whether eval.Normal state is persisted to the database and returned by get methods.
+	doNotSaveNormalState bool
+	// maxStateSaveConcurrency controls the number of goroutines (per rule) that can save alert state in parallel.
+	maxStateSaveConcurrency int
+}
+
+func NewSyncStatePerisiter(log log.Logger, store InstanceStore, doNotSaveNormalState bool, maxStateSaveConcurrency int) StatePersister {
```

Comment:
Rather than bool flags for config, can we instead just accept a `ManagerCfg` and read the settings we need off there?

I realize this gets put into the ManagerCfg which is a recursive dependency, but check my other comment on how to prevent that.

Reply:
(This also implies keeping MaxStateSaveConcurrency on ManagerCfg)

Reply:
Done